Classes/EventStore/Storage/Doctrine/Factory/ConnectionFactory.php

```diff
@@ -41,25 +42,36 @@ class ConnectionFactory
      */
     protected $defaultFlowDatabaseConfiguration;
 
+    /**
+     * A runtime cache for the created connections
+     * (same options == same connection instance)
+     *
+     * @var array
+     */
+    private $connections = [];
+
     /**
      * @param array $options
      * @return Connection
+     * @throws DBALException
      */
     public function create(array $options)
     {
-        $config = new Configuration();
+        $cacheIdentifier = md5(http_build_query($options));
+        if (array_key_exists($cacheIdentifier, $this->connections)) {
+            return $this->connections[$cacheIdentifier];
+        }
         $connectionParams = $options['backendOptions'] ?? [];
         $connectionParams = Arrays::arrayMergeRecursiveOverrule($this->defaultFlowDatabaseConfiguration, $connectionParams);
 
-        $connection = DriverManager::getConnection($connectionParams, $config);
+        $this->connections[$cacheIdentifier] = DriverManager::getConnection($connectionParams);
```

Comment:
`$connection = $this->connections[..] = DriverManager::... ` would avoid the below cumbersome accessors and keep the cache logic a tiny bit more concise

Reply:
```suggestion
        $connection = $this->connections[$cacheIdentifier] = DriverManager::getConnection($connectionParams);
```

Reply:
I had to use the new "suggest change" feature but it doesn't work here :(
Good idea, though!

Reply:
Like this?